From ed1d49466329ffa6cfda7f6fba923167c5077058 Mon Sep 17 00:00:00 2001
From: Nawal Kishor <nkishor@marvell.com>
Date: Wed, 22 May 2024 17:10:25 +0530
Subject: [PATCH 866/955] test/mbuf: fix pkt_len after splitting chain

In mbuf_autotest, while splitting the m0 chain, pkt_len was not reduced
accordingly. Fixes the pkt_len by subtracting pkt_len of last packet
detached.

Fixes: efc6f9104c8 ("mbuf: fix reset on mbuf free")

Signed-off-by: Nawal Kishor <nkishor@marvell.com>
Change-Id: I8fc26684bbbdec0f7826bad0e7d1a32080fe3703
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/dataplane/dpdk/+/128244
Reviewed-by: Ashwin Sekhar T K <asekhar@marvell.com>
Reviewed-by: Jerin Jacob <jerinj@marvell.com>
Tested-by: Jerin Jacob <jerinj@marvell.com>
---
 app/test/test_mbuf.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/app/test/test_mbuf.c b/app/test/test_mbuf.c
index 220cc71e8554e..cddb0ff30643f 100644
--- a/app/test/test_mbuf.c
+++ b/app/test/test_mbuf.c
@@ -2741,6 +2741,7 @@ test_nb_segs_and_next_reset(void)
 
 	/* split m0 chain in two, between m1 and m2 */
 	m0->nb_segs = 2;
+	m0->pkt_len -= m2->pkt_len;
 	m1->next = NULL;
 	m2->nb_segs = 1;
 
-- 
2.25.1

